### DISCUSSION TITLE
Web incompatible images should not be limited to preview-size when zoomed

### DISCUSSION THREAD

**Discussion by wnagrodzki** [immich-app/immich#14173] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I am missing a feature that would generate JPEG/WebP versions for HEIC images in resolution better than preview provides (System Settings > Preview Settings > Resolution), ideally in the resolution the original file has. This version should be used when zooming, similarly how originals are loaded when user zooms. This is currently not the case from what I see [in the source code](https://github.com/immich-app/immich/blob/d4ca7d0075f8374ed9e15398989b9a3440083d26/web/src/lib/components/asset-viewer/photo-viewer.svelte#L42), originals are not loaded for web incompatible images (which makes sense since HEIC is not widely supported).

The current immich web-app behavior for JPEG:
Thumbnail is displayed on timeline. Users clicks o it and preview is displayed. User zooms is, original is loaded.

The current immich web-app behavior for HEIC:
Thumbnail is displayed on timeline. Users clicks o it and preview is displayed. User zooms is, preview stays, original is not loaded.

I am aware that this would create JPEG/WebP copies of HEIC files in original resolution and eat up disk space, but this is a price I am willing to pay for having full resolution available when zooming HEIC images. This could be an opt-in setting.

I am using an external library in read-only mode to seed files into Immich.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by mertalev** [comment]
The resolution you set in the settings is effectively the maximum size of the thumbnail or preview. It will downscale images above that resolution, but won’t upscale smaller images.

**Comment by wnagrodzki** [comment]
@mertalev Thank you for the input, hopefully updated feature description is more clear now. The problem I am trying to solve is that I have 24MP HEIC image, immich generates preview in 1440p resolution and there is no way to load the original file due to HEIC format. I would like to have the option to convert 24MP HEIC into 24MP JPEG so it could be loaded as original file when zooming. What is important to me is to keep original HEIC images intact during the process.

This resembles video-transcoding settings, where you can convert HEVC into h264 and keep original resolution.

**Comment by wnagrodzki** [comment]
This has been implemented in https://github.com/immich-app/immich/releases/tag/v1.131.0
>View HEIC, RAW and other image formats in full resolution in the web app

@mertalev Thank you
